File: src/monitor/__tests__/Parser.test.ts
import { Parser } from '../Parser';

test('parsing raw data to captur object', () => {

    // setup
    const parser = new Parser();

    // execute
    const capture = parser.parse('A8000060001410243028105065904');

    // verify
    expect(capture.elapsedTime).toBe(6);
    expect(capture.distance).toBe(14);
    expect(capture.time500m).toBe(163);
    expect(capture.watts).toBe(105);
    expect(capture.cals).toBe(659);
    expect(capture.level).toBe(4);
});

File: internal/serialmonitor/parser_test.go
package serialmonitor

import (
	"reflect"
	"testing"
)

func Test_parse(t *testing.T) {
	line := "A8000060001410243028105065904"
	want := capture{
		elapsedTime:      6,
		distance:         14,
		time500m:         163,
		strokesPerMinute: 28,
		watts:            105,
		cals:             659,
		level:            4,
	}
	if got := parse(line); !reflect.DeepEqual(got, want) {
		t.Errorf("parse() = %v, want %v", got, want)
	}
}
